### PR TITLE
Add script for building image based on revpi image

### PR DESCRIPTION
**Pull request by zhan-kunbus**
Signed-off-by: Zhi Han <z.han@kunbus.com>

### CODE REVIEW

File: install_debs_into_image.sh [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		-h|--help) usage ; exit 0;;
+		*) shift; break ;;
+	esac
+done
+
+if [ "$MINIMG" != "1" ]; then
+	echo "All additional applications will be built into the given image."
+else
+	echo "Only a reduced application set will be built into the given image."
+fi
```
Comment (nbuchwitz) [inline install_debs_into_image.sh:66]: I think we can omit the whole minimg handling in this script, as it is not used at all.

File: install_debs_into_image.sh [hunk 2]
```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+# customize raspbian image for revolution pi
+
+usage () {
+	echo 'Usage: customize_image.sh [-m, --minimize | -h, --help] <image>
+  -m, --minimize	Install only software that is necessary for basic operation (eg. Pictory and other RevPi tools)
```
Comment (nbuchwitz) [inline install_debs_into_image.sh:6]: I think we can omit the whole minimg handling in this script, as it is not used at all.

File: install_debs_into_image.sh [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+if ! MYOPTS=$(getopt -o mh --long minimize,help -- "$@"); then
+	usage;
+	exit 1;
+fi
+eval set -- "$MYOPTS"
+
+# extract options and their arguments into variables.
+while true ; do
+	case "$1" in
+		-m|--minimize) MINIMG=1 ; shift ;;
```
Comment (nbuchwitz) [inline install_debs_into_image.sh:56]: I think we can omit the whole minimg handling in this script, as it is not used at all.

**Review by nbuchwitz — CHANGES_REQUESTED** [review]

**Comment by nbuchwitz** [comment]
@zhan-kunbus we should also mention the usage and intention of this script in the README

File: install_debs_into_image.sh (hunk 3 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:56]: good idea

File: install_debs_into_image.sh [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+# from ld.so:
+#   ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so'
+#   from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
+mv "$IMAGEDIR/etc/ld.so.preload" "$IMAGEDIR/etc/ld.so.preload.bak"
+
+# customize settings
+echo `basename "$1"` > "$IMAGEDIR/etc/revpi/image-release"
+
+chroot "$IMAGEDIR" apt-get update
+chroot "$IMAGEDIR" apt-get -y install revpipyload
```
Comment (zhan-kunbus) [inline install_debs_into_image.sh:157]: @nbuchwitz  should the installation of revpipyload be replaced with package list from debs-to-download? or just remove it and apt-get update to eliminate the dependency of internet connectivity ?  or just keep it as it now?

File: install_debs_into_image.sh (hunk 1 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:66]: good idea, I will remove it

File: install_debs_into_image.sh (hunk 2 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:6]: good idea, I will remove it

File: install_debs_into_image.sh (hunk 4 above)
Comment (nbuchwitz) [reply install_debs_into_image.sh:157]: Yes, please use debs-to-download

File: README.md [hunk 5]
```diff
@@ -35,6 +35,15 @@ For an image with all additional components (like NodeRed, logi-rts and Teamview
 
 `customize_image.sh <raspbian-image>`
 
+
+### Install debian packages into Revolution Pi Image
+
+To build an image with only update of local packages and Revolution Pi packages, you can get it by following command with a released revolution pi image:
```
Comment (nbuchwitz) [inline README.md:41]: My proposal, which also includes a quick how-to add a package:

> If you would like to modify an existing image by only installing some packages (from repository or local file), you can use the script `install_debs_into_image.sh`. To add a package you can either add the package name to the file `debs-to-download` or put the package file into the folder `debs-to-install/`. After that you have to invoke the script as following:

File: README.md [hunk 6]
```diff
@@ -35,6 +35,15 @@ For an image with all additional components (like NodeRed, logi-rts and Teamview
...
+
+`install_debs_into_image.sh <revpi-image>`
+
+The script requires the same running environment as previous customizing image
```
Comment (nbuchwitz) [inline README.md:45]: Why? Shouldn't it be enough to have all relevant tools installed and run the script against eg. the buster-latest image?

File: install_debs_into_image.sh [hunk 7]
```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+# customize raspbian image for revolution pi
```
Comment (nbuchwitz) [inline install_debs_into_image.sh:2]: Proposal:
> Customize packages of an existing image for Revolution Pi

File: install_debs_into_image.sh [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+usage () {
+	echo 'Usage: customize_image.sh [-m, --minimize | -h, --help] <image>
```
Comment (nbuchwitz) [inline install_debs_into_image.sh:5]: Remove `--minimize` from usage

**Review by nbuchwitz — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (zhan-kunbus) [reply README.md:45]: I mean the same environment as this:
"The script requires root privileges, an armhf system (RasPi or VM) and internet connectivity.

If no armhf system is available, a crossbuild can be done with the qemu user static tools:

```
sudo apt-get install qemu-user-static binfmt-support
```"

Comment (nbuchwitz) [reply README.md:45]: OK, then I'd opt to remove this line. From the textual flow it should be clear, that the requirements are mandatory for booth scripts.

**Review by nbuchwitz — APPROVED** [review]
LGTM, thanks for your work!

**Pull request merged**